c10/util/TypeCast.h

```diff
@@ -131,7 +131,7 @@ C10_HOST_DEVICE inline void cast_and_store(const ScalarType dest_type, void *ptr
 }
 
 template<>
-inline void cast_and_store<std::complex<float>>(const ScalarType dest_type, void *ptr, std::complex<float> value_) {
+C10_HOST_DEVICE inline void cast_and_store<std::complex<float>>(const ScalarType dest_type, void *ptr, std::complex<float> value_) {
```

Comment:
Hi @iotamudelta, is there any specific warning you are seeing? IIRC when I wrote this piece of code, I did this intentionally because this function shouldn't be called inside a `__device__` or `__device__ __host__` function.

Reply:
We are seeing issues with this with clang (not hcc) which does instantiate for complex. I agree that it's not UB if it's not doing complex.

Reply:
I should add: with a vanilla PyTorch code base.

Reply:
Strange, this could be a problem only if inside the kernel the dispatch has `scalar_t` being any of the complex types, which I don't think there is any.